components/brave_new_tab_ui/components/default/together/index.tsx

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this file,
+* You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import * as Crypto from 'crypto'
+import createWidget from '../widget/index'
+import { getLocale } from '../../../../common/locale'
+
+import {
+  WidgetWrapper,
+  Header,
+  Content,
+  WelcomeText,
+  ActionsWrapper,
+  CallButton,
+  TogetherIcon,
+  StyledTitle,
+  StyledTitleText
+} from './style'
+import { StyledTitleTab } from '../widgetTitleTab'
+import BraveTogetherIcon from './assets/brave-together-icon'
+
+interface Props {
+  showContent: boolean
+  onShowContent: () => void
+}
+
+class Together extends React.PureComponent<Props, {}> {
+
+  getButtonText = () => {
+    return getLocale('togetherWidgetStartButton')
+  }
+
+  renderTitle () {
+    return (
+      <Header>
+        <StyledTitle>
+          <TogetherIcon>
+            <BraveTogetherIcon />
+          </TogetherIcon>
+          <StyledTitleText>
+          {getLocale('togetherWidgetTitle')}
+          </StyledTitleText>
+        </StyledTitle>
+      </Header>
+    )
+  }
+
+  renderTitleTab () {
+    const { onShowContent } = this.props
+
+    return (
+      <StyledTitleTab onClick={onShowContent}>
+        {this.renderTitle()}
+      </StyledTitleTab>
+    )
+  }
+
+  shouldCreateCall = (event: any) => {
+    event.preventDefault()
+
+    Crypto.randomBytes(32, (err, buffer) => {
+      if (!err) {
+        const name = buffer.toString('base64').replace(/\+/g, '-').replace(/\//g, '_').replace(/\=/g, '')
```

Comment:
if we continue to use base64 encoding here (and not a guid which is already url-safe we should use `encodeURIComponent`

Reply:
actually I guess this is a standard for url safe base64, I'd just prefer if there was a standard method to generate it

Reply:
there's various libraries that implement it but i think this is pretty much what they do. 

note that if the encoding changes, the server validation code also needs to change (https://github.com/brave/security/issues/147#issuecomment-628203459).